flake.nix

```diff
@@ -1,46 +1,89 @@
 {
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs";
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
     utils.url = "github:numtide/flake-utils";
-    naersk.url = "github:nix-community/naersk";
+    naersk = {
+      url = "github:nix-community/naersk";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
   };
 
-  outputs = { self, nixpkgs, utils, naersk }:
-    utils.lib.eachDefaultSystem (system:
-    let
-      pkgs = nixpkgs.legacyPackages."${system}";
-      naersk-lib = naersk.lib."${system}";
-    in rec 
-    {
-      packages.amethyst = naersk-lib.buildPackage {
-        pname = "ame";
-        root = ./.;
-        nativeBuildInputs = with pkgs; [
-          openssl
-          sqlite
-          pkg-config
-        ];
-      };
-      
-      packages.default = packages.amethyst;
-
-      apps.amethyst = utils.lib.mkApp {
-        drv = packages.amethyst;
-      };
-      
-      apps.default = apps.amethyst;
-
-      devShells.default = pkgs.mkShell {
-        nativeBuildInputs = with pkgs; [
-          rustc
-          cargo
-          rustfmt
-          cargo-audit
-          clippy
-          openssl
-          sqlite
-          pkg-config
-        ];
-      };
-    });
+  outputs = {
+    self,
+    nixpkgs,
+    utils,
+    naersk,
+    fenix,
+  }:
+    utils.lib.eachDefaultSystem (system: let
+        pkgs = nixpkgs.legacyPackages."${system}";
+        toolchain = with fenix.packages."${system}";
+          combine [
+            minimal.rustc
+            minimal.cargo
+            targets.x86_64-pc-windows-gnu.latest.rust-std
+            targets.x86_64-unknown-linux-gnu.latest.rust-std
+          ];
+        naersk-lib = naersk.lib."${system}".override {
+          cargo = toolchain;
+          rustc = toolchain;
+        };
+      in rec
+      {
+        packages.amethyst = naersk-lib.buildPackage {
+          pname = "Amethyst";
+          root = ./.;
+        };
+
+        packages.amethyst-win = naersk-lib.buildPackage {
+          pname = "Amethyst";
+          root = ./.;
+          strictDeps = true;
+          depsBuildBuild = with pkgs; [
+            pkgsCross.mingwW64.stdenv.cc
+            pkgsCross.mingwW64.windows.pthreads
+          ];
+          nativeBuildInputs = with pkgs; [
+            (
+              if system == "x86_64-linux"
+              then wineWowPackages.stable
+              else hello
+            )
+          ];
+          CARGO_BUILD_TARGET = "x86_64-pc-windows-gnu";
+          CARGO_TARGET_X86_64_PC_WINDOWS_GNU_RUNNER = pkgs.writeScript "wine-wrapper" ''
+            export WINEPREFIX="$(mktemp -d)"
+            exec wine64 $@
+          '';
```

Comment:
Is the wine stuff really needed? I doubt that anyone will be compiling amethyst for windows.

Reply:
I'd like to keep it for testing building on Windows, specifically for MSYS2 compatibility (I want to use Amethyst on Windows, damn it! :laughing:)

Reply:
Oh alright, if it doesn't break anything it's fine ig